src/main/scala/sigmastate/serialization/MethodCallSerializer.scala

```diff
@@ -16,6 +17,8 @@ case class MethodCallSerializer(opCode: Byte, cons: (Value[SType], SMethod, Inde
       assert(mc.args.nonEmpty)
       w.putValues(mc.args)
     }
+    w.putUByte(mc.typeSubst.size)
+    mc.typeSubst.foreach { case (typeIdent, tpe) => w.putType(typeIdent).putType(tpe) }
```

Comment:
We don't need to save type idents because they can be restored from Method.stype instance, where we already have a Seq of them. We just need to serialize tpes in the same order.

Reply:
@aslesarenko Interesting idea. I have two questions: 
1. We have Seq of type params `SFunc`. What about `Option.get` which `stype` is `STypeIdent`?
2. Would not it be too fragile, when simple type params reordering in method declaration would break deserialization of existing data?

Reply:
1) We have `SGenericType` and by mistake SOption is not inherited from it.
2) I don't think we need to support changing order paying such a big price at blockchain scale :)

Reply:
Done in 
https://github.com/ScorexFoundation/sigmastate-interpreter/pull/400/commits/391cec0bc40fedaf2229ec4e5dafc52093f7f9c7